pkg/nsx/services/inventory/builder.go

```diff
@@ -448,6 +410,53 @@ func (s *InventoryService) BuildService(service *corev1.Service) (retry bool) {
 	return
 }
 
+// determineServiceStatus determines the complete service status with all available information in one go
+func (s *InventoryService) determineServiceStatus(podIDs []string, hasAddr, hasSelector bool, annotations map[string]string, networkErrors *[]common.NetworkError, uniqueErrors map[string]bool) (string, string) {
+	status := InventoryStatusUp
+	netStatus := NetworkStatusHealthy
+
+	hasAnnotationError := false
+	for key, value := range annotations {
+		if util.Contains(ServiceNCPErrors, key) {
+			hasAnnotationError = true
+			errorMessage := key + ":" + value
+			if !uniqueErrors[errorMessage] {
+				uniqueErrors[errorMessage] = true
+				*networkErrors = append(*networkErrors, common.NetworkError{
+					ErrorMessage: errorMessage,
+				})
+			}
+		}
+	}
+
+	if len(podIDs) > 0 {
```

Comment:
Just to double confirm if those cases are expected:
- When a service without Selector is created, as it does not get podIDs/address and also not set Selector, it will be shown as UP. And after the address is configured, the service status will change from UP to Unknown?
- If we get the podIDs, the service will always be shown as UP even if there is error annotation on the service. I'm not sure if there will be a case like this, if not, it should be fine.

Reply:
def _convert_service(self, key, service):
    deleted = []
    added = []
    project_id = None
    retry = False
    network_errors, network_status = self.update_network_errors(
        key, self._get_external_id(key, service))

status depends on ep status
network_status depends on anno error